.github/workflows/run_unit_tests.yaml

```diff
@@ -0,0 +1,30 @@
+name: Publish
+
+on:
+  pull_request:
+    branches: [ develop ]
+
+jobs:
+
+  test:
+    name: Unit tests
+    runs-on: macOS-latest
```

Comment:
Be careful of using the macOS runners, it will multiply the consumed minutes by **10** times.
For eg. if the build took 8min it will cost you 80min from your 2000min quota.
Theoretically, it should be fine, just keep it in mind so you don't exceed it.

Reply:
More details: https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#minute-multipliers

Reply:
I just wanted to test which one is faster, for now ubunto is faster than macOs, therefore i rollnacked to ubunto.